dico_toolbox/graph.py

```diff
@@ -62,27 +92,135 @@ def stack_buckets(vertex, labels=["aims_ss", "aims_bottom", "aims_other"]):
     except:
         stack = None
         print(vertex_bcks)
-    
+
     return stack
 
 
-def get_bucket_from_graph_by_suclus_name(graph, sulcus_name, ICBM2009c, labels=["aims_ss", "aims_bottom", "aims_other"]):
-    """
-    Get and stack all buckets in the graph corresponding to sulcus name and labels.
-    
-    if ICBM2009c is True, the buckets are transformed into the ICBM2009c Template coordinates.
+def list_buckets(graph, key=None, needed=None, return_keys=None, defaults=None, transform=None, bck_types=BUCKETS_TYPES):
+    """ Stack all buckets of the graph that correspond to needed key values and bucket types.
+        Also return values of each vertex for the specified keys.
+
+        Parameters
+        ==========
+        graph: soma.aims.Graph | str
+            AIMS Cortical Graph
+
+        key: str (opt.)
+            Select vertices that have key property.
+
+        needed: list | single value (opt.)
+            Select vertices that have key value equal to one of the needed values.
+
+        return_keys: list | str | None (opt.)
+            List of returned key values. If return_keys is a string, values are
+            returned in a list otherwise it is done in a dictionnary.
+
+        defaults: list | str | None (opt.)
+            When return_keys is not None, if a key is not defined in a vertex, 
+            defaults are used to set up the value. If defaults is a single value,
+            the value is used for all the keys. Default is None.
+
+        transform: "ICBM2009c" | "Talairach" | None (opt.)
+            Apply known transformation. Two spaces are available: "Talairach" 
+            and "ICBM2009c"(refers to the MNI space used by BrainVISA).
+            None value skip the transformation step.
+            Default is None.
+
+        bck_types: list of str (opt.)
+            Bucket keys used to list points. 
+            Default is ['aims_ss', 'aims_other', 'aims_bottom'].
+
+        Return
+        ======
+        stacked buckets and values dict (or list if return_keys is a string) if asked.
```

Comment:
Return a Tuple (buckets, other_values). If return_keys, other_values is none.
(cfr. le commentaire sur return)